client/src/components/admin/Modal/ImgUploadModal.js

```diff
@@ -35,78 +41,27 @@ export default function ImgUploadModal({userName,type, buttonName}) {
         try {
             dispatch(uploadImageServer({
                 method: 'POST',
-                img : JSON.stringify({ data: base64EncodedImage }),
-                userName : userName,
-                dataFor : type,
-                headers: { 'Content-Type': 'application/json' }}))
-
-
-
-
+                img: JSON.stringify({ data: base64EncodedImage }),
+                userName: userName,
+                dataFor: type,
+                headers: { 'Content-Type': 'application/json' }
+            }))
             setFileInputState('');
-            setOpen(false)
         } catch (err) {
             console.error(err);
         }
     };
 
-
-    function getModalStyle() {
-        const top = 50
-        const left = 50
-        return {
-            top: `${top}%`,
-            left: `${left}%`,
-            transform: `translate(-${top}%, -${left}%)`,
-        };
-    }
-
-    const useStyles = makeStyles(theme => ({
-        modal: {
-            display: 'flex',
-            alignItems: 'center',
-            justifyContent: 'center',
-        },
-        paper: {
-            position: 'absolute',
-            width: 450,
-            backgroundColor: theme.palette.background.paper,
-            boxShadow: theme.shadows[5],
-            padding: theme.spacing(2, 4, 3),
-        },
-    }));
-
     const classes = useStyles();
-    const [modalStyle] = useState(getModalStyle);
-    const [open, setOpen] = useState(false);
-
-    const handleOpen = () => {
-        setOpen(true);
-    };
-
-    const handleClose = () => {
-        setOpen(false);
-    };
 
     return (
         <div>
-            <Button variant="contained" id="edit-poster" color="primary" onClick={handleOpen}>
-                {buttonName}
-            </Button>
-
-            <Modal
```

Comment:
What about this modal?

Reply:
The modal didn't had any function as such, other than providing the button for choosing a file from the file system. So i removed it and added the functionality directly into the edit button.